### PR TITLE
Bug 2037075: Add Support for Build CSI Volume Sources

### PR DESCRIPTION
**Pull request by jkhelil**
https://issues.redhat.com/browse/BUILD-275

### CODE REVIEW

**Comment by gabemontero** [comment]
proxy failure is unrelated to these changes

if I read tide correctly, the OCM repo does not appear to be a no feature freeze repo @jkhelil so you'll need to open a bz and associated it with this PR to get this in, unless @adambkaplan thinks he can change that 

otherwise, I'll 

/lgtm

and 

/assign @adambkaplan 

for the approve label

**Comment by adambkaplan** [comment]
/hold

This PR needs https://github.com/openshift/openshift-apiserver/pull/264 to land first.

**Comment by adambkaplan** [comment]
/retitle Bug 2037075: Add Support for Build CSI Volume Sources

File: pkg/build/controller/strategy/util.go
Comment (adambkaplan) [inline pkg/build/controller/strategy/util.go:720]: I would reference the feature gate name here. Admins can enable tech preview on the cluster to do this, OR provide the feature gate name with their own custom feature set:

```suggestion
				return fmt.Errorf("csi volumes require the BuildCSIVolumes feature gate to be enabled")
```

**Review by adambkaplan — COMMENTED** [review]
A suggestion wrt the error message we report if the feature gate is not enabled. Otherwise looks good.

**Comment by jkhelil** [comment]
/assign @adambkaplan 
/cc @gabemontero 
Can you approve please ?

**Review by adambkaplan — APPROVED** [review]
/approve

**Comment by adambkaplan** [comment]
/lgtm

Still keeping the hold until the openshift-apiserver merges.

**Comment by gabemontero** [comment]
apiserver PR merged

/hold cancel

**Pull request merged**
